src/main/java/com/elytradev/concrete/inventory/ConcreteItemStorage.java

```diff
@@ -155,12 +156,10 @@ public boolean hasValidator(int slot) {
 	 */
 	@Nonnull
 	public Predicate<ItemStack> getValidator(int slot) {
-		if (validators.size()<=slot) return Validators.ANYTHING;
-		return validators.get(slot);
+		return hasValidator(slot) ? validators.get(slot) : Validators.ANYTHING;
 	}
 	
 	public boolean getCanExtract(int slot) {
-		if (extractMask.length<=slot) return false;
-		return extractMask[slot];
+		return hasValidator(slot) && extractMask[slot];
```

Comment:
I'm a bit confused by that. Shouldn't it be `extractMask.length > slot`? Let's say `extractMask.length` is 1 and `slot` is 1 or 2. Doing `extractMask[slot]` would throw an `ArrayIndexOutOfBoundsException`.